### ISSUE TITLE
RE: Using Adafruit_GPIO.I2C and/or Using Adafruit_I2C?

### ISSUE THREAD

**Issue by silver2row** [adafruit/adafruit-beaglebone-io-python#237]
Hello...

uname -a:

    Linux beaglebone 4.9.78-ti-r94 #1 SMP PREEMPT Fri Jan 26 21:26:24 UTC 2018 armv7l GNU/Linux

python:

    Python 2.7.13

As everyone is aware...I am using the Motor Bridge Cape w/ Adafruit_BBIO, version 1.0.10.

Now...on the Seeed-Studio, GitHub.com page for the Motor Bridge Cape, there is a MotorBridge.py library I "must" follow to get my "below par" software running on the BBG w/ Debian Stretch 4.9.x.

So,

I have changed, b/c of the mvduin fellow's advice on MotorBridgeCape number 6 issue, some software on the MotorBridge.py library to suit the needs of the Adafruit_BBIO, version 1.0.10 library.

Now...supposedly:

    import Adafruit_GPIO.I2C as I2C
    import Adafruit_BBIO.GPIO as GPIO
    import time
    Reset = "P9_23"
    MotorBridge = I2C.Device(0x4b, 2)
    GPIO.setup(Reset, GPIO.OUT)
    
This is how I should have changed my MotorBridge.py library and I listened to issue number 6.

I am sure you have a ton of time away from this subject. If you can try to reply to let me in on what is transpiring on your side, I can then work w/ the Adafruit_BBIO library some more.

Here is my glitch/error:

    Traceback (most recent call last):
      File "mbcUse.py", line 1, in <module>
        import MotorBridge
      File "/home/debian/MotorBridgeCapeforBBG_BBB/BBG_MotorBridgeCape/MotorBridge.py", line 33, in 
    <module>
        import Adafruit_GPIO.I2C as I2C
    ImportError: No module named Adafruit_GPIO.I2C

Seth

P.S. If you know what happened so far in these two, separate libraries, please let me know.

**Comment by silver2row** [comment]
Hello,

I then cloned adafruit/adafruit-python-gpio due to mvduin's ideas and left the above change to MotorBridge.py. 

I ran my mbcUse.py software found at https://github.com/silver2row/MotorBridgeCapeforBBG_BBB/blob/silver2row-patches/BBG_MotorBridgeCape/mbcUse.py.

This is the error I receive now:

    Traceback (most recent call last):
      File "mbcUse.py", line 13, in <module>
        motor.DCMotorInit(1, 1000)
      File "/home/debian/MotorBridgeCapeforBBG_BBB/BBG_MotorBridgeCape/MotorBridge.py", line 279, 
    in DCMotorInit
        WriteOneWord(CONFIG_TB_PWM_FREQ,Frequency)
      File "/home/debian/MotorBridgeCapeforBBG_BBB/BBG_MotorBridgeCape/MotorBridge.py", line 221, 
    in WriteOneWord
        MotorBridge.writeList(WriteMode,data)
      File "build/bdist.linux-armv7l/egg/Adafruit_GPIO/I2C.py", line 129, in writeList
      File "/usr/local/lib/python2.7/dist-packages/Adafruit_PureIO-0.2.1-
    py2.7.egg/Adafruit_PureIO/smbus.py", line 274, in write_i2c_block_data
        self._device.write(data)
    IOError: [Errno 121] Remote I/O error

Although expansive, this sums up the error from the MotorBridge.py library change and Adafruit_BBIO/Adafruit_GPIO, I think.

Seth

P.S. If you have any ideas, please let me know. I will keep on it and change things accordingly. Thank you again for your service.

**Comment by silver2row** [comment]
Hello,

I noticed some differences in my system. Using mvduin's show-pins utility, I ended up w/ these two pastebin.com pastes:

- https://pastebin.com/95ntRftF

- https://pastebin.com/zV4Pf8NZ

These show my current configuration on two, separate boards each running different Debian images and different kernels.

Seth

P.S. If you notice P9_19 and P9_20 on both pastebin.com pastes, you will see how they differ. If you have time or know if this even matters, let me know. Thank you again for understanding my concern.
